### PR TITLE
Add basic support for strings according to the Molang documents on Mi…

### PR DESCRIPTION
**Pull request by rbrick**
…crosoft:

https://learn.microsoft.com/en-us/minecraft/creator/reference/content/molangreference/examples/molangconcepts/molangintroduction#strings

### CODE REVIEW

File: src/main/java/com/eliotlash/molang/lexer/Lexer.java [hunk 1]
```diff
@@ -79,6 +85,12 @@ private void scanToken() {
 		}
 	}
 
+
+	private void eatString() {
+		while (advance() != '\'') {}
```
Comment (bernie-g) [inline src/main/java/com/eliotlash/molang/lexer/Lexer.java]: We should add some error handling for when you're missing a closing quote. Also, should we support double quotes?

File: src/main/java/com/eliotlash/molang/variables/ExecutionContext.java [hunk 2]
```diff
@@ -59,6 +61,9 @@ public ExecutionContext(Evaluator evaluator) {
         registerFunction("math", new Lerp("lerp"));
         registerFunction("math", new LerpRotate("lerprotate"));
         registerFunction("math", new Random("random"));
+
+        registerFunction("system", new Print("print"));
+        registerFunction("strings", new StrEquals("equals"));
```
Comment (bernie-g) [inline src/main/java/com/eliotlash/molang/variables/ExecutionContext.java]: Let's do `string.equals` for consistency

File: src/test/java/com/eliotlash/molang/FunctionsTest.java [hunk 3]
```diff
@@ -91,11 +91,14 @@ void testMiscFunctions() throws Exception {
         assertEquals(10.0, evaluate("math.abs(-10)"));
         assertEquals(10.0, evaluate("math.abs(10)"));
 
-        assertEquals(Math.E, evaluate("math.exp(1)"));
+        assertEquals( 2.7182818284590455, evaluate("math.exp(1)"));
```
Comment (bernie-g) [inline src/test/java/com/eliotlash/molang/FunctionsTest.java]: ? lol

**Review by bernie-g — APPROVED** [review]
Looks great!!

File: src/test/java/com/eliotlash/molang/FunctionsTest.java (hunk 3 above)
Comment (rbrick) [reply src/test/java/com/eliotlash/molang/FunctionsTest.java]: There is a discrepancy in the floating point accuracy between the constant `Math.E` & the result of `Math.exp(1)` on some JVM implementations, at least on macOS.


Constant E: `2.718281828459045`


Results on OpenJDK & Zulu 8, 16, & 17, and IntelliJs version of JDK (v11), the result of `Math.exp(1)` == `2.7182818284590455`

Results on GraalVM 17: `2.718281828459045`


Given that this seems to be up to the implementation of the JDK itself and that the result is not reliably reproducible across multiple versions of the JDK, I think we should just remove this assert.

File: src/main/java/com/eliotlash/molang/lexer/Lexer.java (hunk 1 above)
Comment (rbrick) [reply src/main/java/com/eliotlash/molang/lexer/Lexer.java]: Added an error for unclosed strings in latest commit

File: src/main/java/com/eliotlash/molang/variables/ExecutionContext.java (hunk 2 above)
Comment (rbrick) [reply src/main/java/com/eliotlash/molang/variables/ExecutionContext.java]: Changed to string equals & added a string.equalsIgnoreCase

File: src/test/java/com/eliotlash/molang/FunctionsTest.java (hunk 3 above)
Comment (FxMorin) [reply src/test/java/com/eliotlash/molang/FunctionsTest.java]: I would just use `math.exp(1)` as it will give the same result

Comment (rbrick) [reply src/test/java/com/eliotlash/molang/FunctionsTest.java]: Good point! I switched to using `Math.exp(1)`, and tried resolved merge conflicts

**Pull request merged**